.github/workflows/neard_assertion_binary.yml

```diff
@@ -0,0 +1,49 @@
+name: Neard Assertion binary release
```

Comment:
we could absolutely do this except for different triggers:
regular builds should happend on pushes on master, releases or prereleases
assertion should happen on a period basis (every 8 hours)
nightly should be triggered on demand only. 

Open to suggestions on how to handle these with a matrix if possible

Reply:
I would perhaps follow what `wasmtime` does and maintain a job that generates the matrix depending on the triggers. To do that you can use [job outputs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs) to pass the generated matrix to the actual build job.

Here’s an example:

https://github.com/near/wasmtime/blob/d2b43ee524c7b5ae002a90d762c68de2db2ae773/.github/workflows/main.yml#L112-L133

which produces e.g. `build-matrix` which is then used in the build job: 

https://github.com/near/wasmtime/blob/d2b43ee524c7b5ae002a90d762c68de2db2ae773/.github/workflows/main.yml#L686

In particular the [`github` context contains information about the trigger](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)

Reply:
And for the manually triggered jobs you would need an additional input to specify the release type to run the workflow for?

Reply:
I'd prefer to avoid over-complicated code in favour of simple, small and easy to follow pieces.
This way we also have a more granular control over the runs.

No need for release type on these runs. Assertion is always on master, except when manually triggered and the binary will go into the branch name folder ${BRANCH}/latest-assertions

Reply:
I would like to give a try to @nagisa 's suggestion. But feel free to proceed as is.

Reply:
I'll look into this although not a fan of fancy complex code.
For now the task #1 is to move missing pipelines out of buildkite

Reply:
Would it be very bad to just build the assertions build on each push, especially considering over the last 30 days the merge queue merged on average only slightly more than one PR per 8 hours?
For nightly, that seems never used from what I read on zulip, so we can either remove it, or leave it but add an `if` clause that makes the job a no-op if it’s being run from a master push and is the nightly matrix element; without any need for complex matrix computations as job outputs.

Reply:
Actually there’s an even simpler solution: [matrix excludes can apparently be enough to do it](https://github.com/orgs/community/discussions/26253#discussioncomment-3250989)